lib/ProductOpener/Web.pm

```diff
@@ -378,4 +379,33 @@ sub get_countries_options_list ($target_lc, $exclude_world = 1) {
 	return \@countries_list;
 }
 
+=head2 add_params( $url, $params )
+
+Adds parameters to a URL while properly handling the existing query string.
+
+=head3 Arguments
+
+=head4 $url - The URL to which parameters need to be added.
+
+=head4 $params - The parameters to be added in the format "key=value".
+
+=head3 Return value
+
+The modified URL with added parameters.
+
+=cut
+
+sub add_params ($url, $params) {
+
+	# Check if the URL already contains parameters
```

Comment:
It would be good to check if the URL already contains the same parameter we want to add, in which case we want to remove the old one before adding the new one.

Reply:
It would be good to add unit tests for this function with various cases: param already there, other params, no other params etc.

Reply:
> It would be good to check if the URL already contains the same parameter we want to add, in which case we want to remove the old one before adding the new one.

Do we always explicitly limit it to one value when reading the parameters? [cgi](https://metacpan.org/pod/CGI#Fetching-the-value-or-values-of-a-single-named-parameter) can do arrays

Reply:
@stephanegigandet where should the test cases be written .. Is it in a file in tests/unit or should I create a .t file there for the same ?

Reply:
You can create a tests/unit/web.t file.